### ISSUE TITLE
I cannot get CDATA of response field

### ISSUE THREAD

**Issue by mserralta** [mvantellingen/python-zeep#305]
Hi there!

I am not new in Python but I am with SOAP services and mostly with this new module.

I need to consume a resource that the response is send as an XML inside a CDATA in the SOAP Response. Weird. Ugly. But real!

A bit of the response I get is:

```
<ExecuteTransationResponse xmlns="xxx">
         <ExecuteTransationResponse xmlns="yyy">
            <status xmlns="">OK</status>
            <transactions xmlns="">
               <data><![CDATA[<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<QueryResult>
  <Record>
    <NIF>****</NIF>
    <Nombre_Completo>***</Nombre_Completo>
    <Sexo>H</Sexo>
    <Direccion_Completa>****   </Direccion_Completa>
    <Localidad>***</Localidad>
    <Codigo_Postal>****</Codigo_Postal>
    <Telefono_1>***</Telefono_1>
    <Telefono_Profesional />
    <Codigo_Categoria>*</Codigo_Categoria>
    <Categoria>***                  </Categoria>
    <Fecha_Nacimiento>***</Fecha_Nacimiento>
    <Fecha_de_baja_en_compañia />
    <Nacionalidad>***</Nacionalidad>
    <Fecha_de_alta_en_compañia>****</Fecha_de_alta_en_compañia>
  </Record>
  <Record>
 [...]
  </Record>
</QueryResult>]]>
		           </data>
[...]
	          </transactions>
         </ExecuteTransationResponse>
</ExecuteTransationResponse>
```

The response got by zeep module is right in every case except in data attribute which values None.

What am I doing wrong? Is this module designed to handle this kind of response?

Thanks!

**Comment by mserralta** [comment]
I have been digging arount and I think the problem may be in the module zeep/xsd/types.py

When this attribute is parsed it is done throw the method parse_xmlelement of ComplexType.

Here, if I print the text of the element data I got the entire XML from CDATA. But in the code it returns in line 303

    if allow_none and len(elements) == 0 and len(attributes) == 0:                
      return

what makes this None when I got the response in my script.

Is this behaviour correct?

**Comment by mvantellingen** [comment]
Hi,

Thanks for the report. It is unfortunately missing an example script as explained in the documentation (see http://docs.python-zeep.org/en/master/reporting_bugs.html). This is also mentioned when you create the issue on github.

I realise this will result in extra work on your side, but I can really use the extra help.

**Comment by mserralta** [comment]
Hi there!

Sorry for the test missing. I do attach it. 

The values of field "data" inside response has been modified due to confidencial content but it is still equivalent.

In case it may be useful, I just have noted that the suds library can parse correctly this response. But on the other hand, cannot understand the WSDL.

Thanks!

[test.zip](https://github.com/mvantellingen/python-zeep/files/719014/test.zip)

**Comment by mvantellingen** [comment]
Should be fixed (turned out to be a duplicate of #284)

**Comment by mvantellingen** [comment]
I'm closing this ticket for now as it should be fixed. If there are still any issues then please let me know so that I can re-open the issue.
